File: solutions/ruby/pig-latin/2/pig_latin.rb
# frozen_string_literal: true

# module PigLatin
module PigLatin
  def self.translate(phrase) = phrase.gsub(/\w+/) { pig_latin(_1) }

  def self.pig_latin(word)
    return "#{word}ay" if word.match?(/^([aeiou]|xr|yt)/)

    word.sub(/^((?:[^aeiou]?qu|[^aeiouy]+|y))(.*)$/, '\\2\\1ay')
  end
  private_class_method :pig_latin
end

File: solutions/clojure/pig-latin/2/src/pig_latin.clj
(ns pig-latin
  (:require [clojure.string :as str]))

(defn translate [phrase]
  (str/replace phrase
               #"(?<!\S)(?:(?=[aeiou]|xr|yt)|([^aeiou\s]?qu|[^aeiouy\s]+|y))(\S*)"
               "$2$1ay"))
